src/blueprint/azext_blueprint/_params.py

```diff
@@ -0,0 +1,215 @@
+        c.argument('version_id', options_list=['--version'], help='Version of the published blueprint definition.')
+        c.argument('change_notes', help='Version-specific change notes.')
+
+    with self.argument_context('blueprint published delete') as c:
+        c.argument('blueprint_name', help='Name of the blueprint definition.')
+        c.argument('version_id', options_list=['--version'], help='Version of the published blueprint definition.')
+
+    with self.argument_context('blueprint published show') as c:
+        c.argument('blueprint_name', help='Name of the blueprint definition.')
+        c.argument('version_id', options_list=['--version'], help='Version of the published blueprint definition.')
+
+    with self.argument_context('blueprint published list') as c:
+        c.argument('blueprint_name', help='Name of the blueprint definition.')
+
+    with self.argument_context('blueprint published artifact show') as c:
+        c.argument('blueprint_name', help='Name of the blueprint definition.')
+        c.argument('version_id', options_list=['--version'], help='Version of the published blueprint definition.')
+        c.argument('artifact_name', help='Name of the blueprint artifact.')
+
+    with self.argument_context('blueprint published artifact list') as c:
+        c.argument('blueprint_name', help='Name of the blueprint definition.')
+        c.argument('version_id', options_list=['--version'], help='Version of the published blueprint definition.')
+
+    for scope in ['blueprint assignment create', 'blueprint assignment update']:
+        with self.argument_context(scope) as c:
+            c.argument('assignment_name', options_list=['--name', '-n'], help='Name of the blueprint assignment.')
+            c.argument('location', arg_type=get_location_type(self.cli_ctx))
+            c.argument('identity_type', arg_type=get_enum_type(['None', 'SystemAssigned', 'UserAssigned']), arg_group='identity', help='Type of the managed identity.')
+            c.argument('identity_principal_id', arg_group='identity', help='Azure Active Directory principal ID associated with this Identity.')
+            c.argument('identity_tenant_id', arg_group='identity', help='ID of the Azure Active Directory.')
+            c.argument('identity_user_assigned_identities', arg_group='identity', help='The list of user-assigned managed identities associated with the resource. Key is the Azure resource Id of the managed identity.')
+            c.argument('display_name', help='One-liner string explain this resource.')
+            c.argument('description', help='Multi-line explain this resource.')
+            c.argument('blueprint_id', help='ID of the published version of a blueprint definition.')
+            c.argument('parameters', arg_type=parameter_type, help='Blueprint assignment parameter values.')
+            c.argument('resource_groups', help='Names and locations of resource group placeholders.')

```

Comment:
"Names and locations" means it is array?

Reply:
`resource_groups` is a dict, so I used json here. But as we should limit the usage of json, I have changed it to actions and it can accept multiple resource group key=value pairs. Then I turned it into dictionary in the custom method before calling SDK.